Bottom: 75d1c68039c4ba4ee9942e44ab36085e1a8f3009
Top:    3435a8984ad5412ce014d4e14af56f0376f14ad5
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-08 23:51:58 +0100

draw.c: Draw pins, pads and vias (for non-gui HIDs) from DrawLayerGroup()


---

diff --git a/src/draw.c b/src/draw.c
index 3293c1b..e15d8f4 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -366,7 +366,6 @@ PrintAssembly (int side, const BoxType * drawn_area)
 
   gui->set_draw_faded (Output.fgGC, 1);
   DrawLayerGroup (side_group, drawn_area);
-  DrawPPV (side_group, drawn_area);
   gui->set_draw_faded (Output.fgGC, 0);
 
   /* draw package */
@@ -430,8 +429,7 @@ DrawEverything (BoxTypePtr drawn_area)
 
       if (gui->set_layer (0, group, 0))
         {
-          if (DrawLayerGroup (group, drawn_area) && !gui->gui)
-            DrawPPV (group, drawn_area);
+          DrawLayerGroup (group, drawn_area); 
           gui->end_layer ();
         }
     }
@@ -789,7 +787,9 @@ DrawLayerGroup (int group, const BoxType *drawn_area)
     }
   if (n_entries > 1)
     rv = 1;
-  return rv;
+
+  if (rv == 1 && !gui->gui)
+    DrawPPV (group, drawn_area);
 }
 
 static void
